truffle/src/main/java/org/jruby/truffle/core/format/format/FormatIntegerNode.java

```diff
@@ -10,63 +10,200 @@
 package org.jruby.truffle.core.format.format;
 
 import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
-import com.oracle.truffle.api.dsl.Cached;
 import com.oracle.truffle.api.dsl.NodeChild;
 import com.oracle.truffle.api.dsl.NodeChildren;
 import com.oracle.truffle.api.dsl.Specialization;
 import com.oracle.truffle.api.object.DynamicObject;
 import org.jruby.truffle.Layouts;
 import org.jruby.truffle.RubyContext;
 import org.jruby.truffle.core.format.FormatNode;
-import org.jruby.truffle.core.format.printf.PrintfTreeBuilder;
+import org.jruby.truffle.core.format.printf.PrintfSimpleTreeBuilder;
+import org.jruby.util.ConvertBytes;
 
 import java.math.BigInteger;
 import java.nio.charset.StandardCharsets;
 import java.util.Locale;
 
 @NodeChildren({
         @NodeChild(value = "width", type = FormatNode.class),
+    @NodeChild(value = "precision", type = FormatNode.class),
         @NodeChild(value = "value", type = FormatNode.class),
 })
 public abstract class FormatIntegerNode extends FormatNode {
 
     private final char format;
     private final boolean hasSpaceFlag;
     private final boolean hasZeroFlag;
-    private final int precision;
+    private final boolean hasPlusFlag;
+    private final boolean hasMinusFlag;
+    private final boolean hasFSharp;
 
-    public FormatIntegerNode(RubyContext context, char format, boolean hasSpaceFlag, boolean hasZeroFlag, int precision) {
+    public FormatIntegerNode(RubyContext context, char format, boolean hasSpaceFlag, boolean hasZeroFlag, boolean hasPlusFlag, boolean hasMinusFlag, boolean hasFSharp) {
         super(context);
         this.format = format;
         this.hasSpaceFlag = hasSpaceFlag;
         this.hasZeroFlag = hasZeroFlag;
-        this.precision = precision;
+        this.hasPlusFlag = hasPlusFlag;
+        this.hasMinusFlag = hasMinusFlag;
+        this.hasFSharp = hasFSharp;
     }
 
-    @Specialization(
-            guards = {
-                    "!isRubyBignum(value)",
-                    "width == cachedWidth"
-            },
-            limit = "getLimit()"
-    )
-    public byte[] formatCached(int width,
-                               Object value,
-                               @Cached("width") int cachedWidth,
-                               @Cached("makeFormatString(width)") String cachedFormatString) {
-        return doFormat(value, cachedFormatString);
+    @Specialization
+    public byte[] format(int width, int precision, int value) {
```

Comment:
This kind of large method won't compile well, String concatenation, a large switch, etc.
Probably it can be just a boundary since doFormat was already a boundary.